### PR TITLE
Create publish-gem.yml

### PR DESCRIPTION
**Pull request by tsusdere**
Adds workflow to automate publishing the gem to RubyGems

### CODE REVIEW

File: .github/workflows/publish-gem.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
```
Comment (YiMysty) [inline .github/workflows/publish-gem.yml:17]:
```suggestion
          ruby-version: '2.7.3'
```

File: .github/workflows/publish-gem.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7.3'
```
Comment (parkr) [inline .github/workflows/publish-gem.yml:17]: Is there any reason not to release on a later version? At least worthwhile to use 2.7.8 to get the bug and security fixes.

Comment (YiMysty) [reply .github/workflows/publish-gem.yml:17]: 👋 @parkr do you think if we can move to ruby 3.0 for this project? yeh we can move to 2.7.8

File: .github/workflows/publish-gem.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7.8'
+      - name: Build gem
+        run: |
+          gem build github-pages.gemspec
+      - name: Publish
```
Comment (YiMysty) [inline .github/workflows/publish-gem.yml:21]: this should only run when merge into master branch I think?

Comment (tsusdere) [reply .github/workflows/publish-gem.yml:21]: Do we want to instead run the job when we merge into `master` branch?
```yaml
on:
  push:
    branches:
      - main
```

Comment (YiMysty) [reply .github/workflows/publish-gem.yml:21]:
```
on:
  release:
    types: [released]
```

File: .github/workflows/publish-gem.yml (hunk 2 above)
Comment (tsusdere) [reply .github/workflows/publish-gem.yml]: To add seems the same problem with CI can be seen with other repos (regarding Ruby version):
https://github.com/pages-themes/midnight/actions/runs/7391072383/job/20107215490?pr=50

Comment (parkr) [reply .github/workflows/publish-gem.yml]: I think we should go to 3.2 at least, but there might be bugs to fix along the way.

File: Dockerfile [hunk 4]
```diff
@@ -8,7 +8,8 @@ RUN apt-get update \
     make \
     nodejs
 
-RUN gem update --system
+# final version for rubygems-update to supoprt ruby 2.7, consider removing this after upgraded to ruby 3.0
+RUN gem update --system 3.4.22
```
Comment (parkr) [inline Dockerfile:12]: Since you're upgrading in this PR, you can probably remove this.

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -7,8 +7,8 @@ jobs:
     strategy:
       matrix:
         ruby:
-          - 2.6
-          - 2.7
+          - 3.0
+          - 3.3
```
Comment (parkr) [inline .github/workflows/push.yml:11]: Quote these so they become strings when parsed rather than floats.

File: .travis.yml [hunk 6]
```diff
@@ -1,8 +1,7 @@
 language: ruby
 rvm:
-  - 2.5
-  - 2.6
-  - 2.7
+  - 3.0
+  - 3.3
```
Comment (parkr) [inline .travis.yml:4]: I think you can remove this Travis config file entirely.

**Review by parkr — APPROVED** [review]

**Comment by parkr** [comment]
Ruby 3.3 support is added by https://github.com/github/pages-gem/pull/897. That needs to be merged first!

File: Dockerfile.alpine [hunk 7]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:2.7.3-alpine
+FROM ruby:3.3-alpine
```
Comment (parkr) [inline Dockerfile.alpine:1]: I believe CI is attempting to pass the RUBY_VERSION build arg, but here it's always ruby 3.3.

Either remove the RUBY_VERSION arg from CI and script/cibuild-docker or support it here in Dockerfile.

**Review by parkr — APPROVED** [review]

File: lib/github-pages/dependencies.rb [hunk 8]
```diff
@@ -63,7 +63,7 @@ def self.version_report
       require "nokogiri"
 
       {
-        "ruby" => RUBY_VERSION,
+        "ruby" => "3.3",
```
Comment (parkr) [inline lib/github-pages/dependencies.rb:66]: This will show unexpected values for non-Ruby 3.3 CI.

**Review by parkr — APPROVED** [review]

**Pull request merged**